File: CodalityTestHarness/CodifyTestHarness/3. Time Complexity/PermMissingElem.cs
using System.Linq;

namespace CodalityTestLibrary
{
    /// <summary>
    /// Write a function that, given a zero-indexed array A, 
    /// returns the value of the missing element.
    /// </summary>
    public class PermMissingElem
    {
        public int solution(int[] numericArray){

            int minimalReturnValue = 1;

            // defensive code
            if (numericArray.Count() == 0)
            {
                return minimalReturnValue;
            }

            // Get the range from min to max, find exceptions in the array
            return Enumerable.Range(minimalReturnValue, numericArray.Max() + 1)
                                    .Except(numericArray)
                                    .FirstOrDefault();
        }
    }
}

File: Python/3. Time Complexity/PermMissingElem.py
# Write a function that, given a zero-indexed array A, 
# returns the value of the missing element.
class PermMissingElem(object):

    def solution(self, numericArray):
        minValue = 1

        # defensive code
        if len(numericArray) == 0:
            return minValue
        
        # get the known possible values between min (always 1) and max values provided
        arrayRange = list(range(minValue, (max(numericArray) +1)))

        # find the first value in the provided array that isn't in our known range
        return next((value for value in arrayRange if value not in numericArray), minValue)
